WooCommerce/UITestsFoundation/Screens/Login/GetStartedScreen.swift

```diff
@@ -1,40 +1,32 @@
+import ScreenObject
 import XCTest
 
-private struct ElementStringIDs {
-    static let navBar = "WordPress.GetStartedView"
-    static let emailTextField = "Email address"
-    static let continueButton = "Get Started Email Continue Button"
-}
-
-public final class GetStartedScreen: BaseScreen {
-    private let navBar: XCUIElement
-    private let emailTextField: XCUIElement
-    private let continueButton: XCUIElement
-
-    init() {
-        let app = XCUIApplication()
-        navBar = app.navigationBars[ElementStringIDs.navBar]
-        emailTextField = app.textFields[ElementStringIDs.emailTextField]
-        continueButton = app.buttons[ElementStringIDs.continueButton]
-
-        super.init(element: emailTextField)
+public final class GetStartedScreen: ScreenObject {
```

Comment:
I noticed some differences between this class, and, for example, [OrdersScreen](https://github.com/woocommerce/woocommerce-ios/blob/trunk/WooCommerce/UITestsFoundation/Screens/Orders/OrdersScreen.swift):

1. Using getters will eliminate the need to duplicate strings or define them as identifiers elsewhere.
2. `ScreenObject` already has its own `isLoaded`, so I believe it's not necessary to have it in every ScreenObject instance. Is there any reason why you define another one? There's a behavioural difference between them: `isLoaded` from `ScreenObject` will go through every element from getters array (and for `init` only the first will be checked). This was done to speed up the `init` process while keeping more coverage for elements if we want to go for more extensive `isLoaded`. The approach you use revisits this logic.

I'm pasting the example code here, since I think it's better to dump it in full for easier perception:

```swift
public final class GetStartedScreen: ScreenObject {

    private let getStartedButtonGetter: (XCUIApplication) -> XCUIElement = {
        $0.buttons["Get Started Email Continue Button"]
    }

    private let emailAdressTextFieldGetter: (XCUIApplication) -> XCUIElement = {
        $0.textFields["Email address"]
    }

    private var getStartedButton: XCUIElement { getStartedButtonGetter(app) }
    private var emailAdressTextField: XCUIElement { emailAdressTextFieldGetter(app) }

    public init(app: XCUIApplication = XCUIApplication()) throws {
        try super.init(
              expectedElementGetters: [
                  emailAdressTextFieldGetter,
                  getStartedButtonGetter,
              ],
            app: app
        )
    }

    public func proceedWith(email: String) throws -> PasswordScreen {
        emailAdressTextField.enterText(text: email)
        getStartedButton.tap()

        return try PasswordScreen()
    }

    func isEmailEntered() -> Bool {
        emailAdressTextField.value != nil
    }
}
```

Reply:
Updated to use `getters` in https://github.com/woocommerce/woocommerce-ios/pull/8444/commits/56fe6a7f3b5518c3a557f480d758ccf499f8c8ef

Reply:
Thanks for the follow-ups @jostnes!

> We've had a similar conversation [before](https://github.com/wordpress-mobile/WordPress-iOS/pull/17641/files#r765002927) on WPiOS. Currently, the codebase isn't very consistent in the use of getters/no getters (Orders use them but you can also see that Products don't).

True, my main intent with this comment not as much about "inline" VS "define everything on top" styles (sorry for not making this clear), but about not having same strings for locators multiple times in the file, which is not good for maintenance. Both approaches solve this point 👍 

> I ran the tests before and after removing the separate `isLoaded()` methods and found that the test duration is about the same so the test must have already used the `isLoaded()` defined on ScreenObject instead of the one on the screen

Thanks for the extra investigation! This could be it, or maybe the difference in the number of sought-for elements (`1` vs `2-3`) and the number of times the `isLoaded` is used generally is not that big to make a difference 🤷 In any case, I think we can keep it as it is since it's working now 🙂